internal/operators/ocs/ocs_operator.go

```diff
@@ -0,0 +1,82 @@
+package ocs
+
+import (
+	"context"
+
+	"github.com/kelseyhightower/envconfig"
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/internal/operators/api"
+	"github.com/openshift/assisted-service/models"
+	"github.com/sirupsen/logrus"
+)
+
+type ocsOperator struct {
+	log                logrus.FieldLogger
+	ocsValidatorConfig Config
+	ocsValidator       OCSValidator
+}
+
+func NewOcsOperator(log logrus.FieldLogger) *ocsOperator {
+	cfg := Config{}
+	err := envconfig.Process("myapp", &cfg)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+	validator := NewOCSValidator(log.WithField("pkg", "ocs-operator-state"), &cfg)
+	return NewOcsOperatorWithConfig(log, cfg, validator)
+}
+
+func NewOcsOperatorWithConfig(log logrus.FieldLogger, config Config, validator OCSValidator) *ocsOperator {
+	return &ocsOperator{
+		log:                log,
+		ocsValidatorConfig: config,
+		ocsValidator:       validator,
+	}
+}
+
+func (o *ocsOperator) GetType() models.OperatorType {
+	return models.OperatorTypeOcs
+}
+
+func (o *ocsOperator) GetDependencies() []models.OperatorType {
+	return []models.OperatorType{models.OperatorTypeLso}
+}
+
+func (o *ocsOperator) GetClusterValidationID() string {
+	return string(models.ClusterValidationIDOcsRequirementsSatisfied)
+}
+
+func (o *ocsOperator) GetHostValidationID() string {
+	return string(models.HostValidationIDOcsRequirementsSatisfied)
+}
+
+func (o *ocsOperator) ValidateCluster(_ context.Context, cluster *common.Cluster) (api.ValidationResult, error) {
+	status, message := o.ocsValidator.ValidateOCSRequirements(&cluster.Cluster)
+
+	return api.ValidationResult{Status: status, ValidationId: o.GetClusterValidationID(), Reasons: []string{message}}, nil
+}
+
+func (o *ocsOperator) ValidateHost(context.Context, *common.Cluster, *models.Host) (api.ValidationResult, error) {
+	return api.ValidationResult{Status: api.Success, ValidationId: o.GetHostValidationID(), Reasons: []string{}}, nil
+}
+
+func (o *ocsOperator) GenerateManifests(cluster *common.Cluster) (*api.Manifests, error) {
+	manifests, err := Manifests(o.ocsValidatorConfig.OCSMinimalDeployment, cluster.OpenshiftVersion, o.ocsValidatorConfig.OCSDisksAvailable, len(cluster.Cluster.Hosts))
+	return &api.Manifests{Files: manifests}, err
+}
+
+func (o *ocsOperator) GetCPURequirementForWorker(context.Context, *common.Cluster) (int64, error) {
+	return o.ocsValidatorConfig.OCSMinimumCPUCount / o.ocsValidatorConfig.OCSRequiredHosts, nil
```

Comment:
@jakub-dzon, does this perform the host based check? I don't think we require this. 
cc: @nbalacha

Reply:
This is an API to be used in implementation of https://issues.redhat.com/browse/MGMT-4066, where total requirements of OCP and all operators will be compared with hosts' capacity. 
Do you think it should return "0"?

Reply:
currently, we are not performing any host based validations for OCS. So maybe it can return 0.

Reply:
:+1:

Reply:
Done for both cpu and RAM.